pdsfile/pdsfile_test_helper.py

```diff
@@ -40,21 +40,48 @@ def read_or_update_golden_copy(data, path, update):
 
     path = Path(path)
     # Create the golden copy by using the current output
-    if update or not path.exists():
+    if not path.exists():
         # create the directory to store the golden copy if it doesn't exist.
         os.makedirs(os.path.dirname(path), exist_ok=True)
+        # write the output to the file.
+        write_data_to_file(data, path)
+        return 0
 
-        # write the associated_abspaths output to the file.
-        with open(path, 'w') as f:
-            pprint.pp(data, stream=f)
-        print('\nCreate the golden copy', path)
+    # If the update param is given, compare the file first to make sure we don't write
```

Comment:
Why do we need to do this? If the output is sorted properly, writing a new output file with identical content is fine.